drivers/flash/Kconfig.da1469x

```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2022 Renesas Electronics Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_FLASH_DA1469X
+	bool "DA1469x flash driver"
+	depends on SOC_FAMILY_DA1469X
+	select FLASH_HAS_PAGE_LAYOUT
+	select FLASH_HAS_DRIVER_ENABLED
+	select MPU_ALLOW_FLASH_WRITE if ARM_MPU
+	default y
```

Comment:
make this based on `default $(dt_compat_enabled,$(DT_COMPAT_RENESAS_DA1469X_FLASH_CONTROLLER))`